### PR TITLE
building the binary inside a docker host instead of depending on the host os/arch

### PR DESCRIPTION
**Pull request by nicob87**
Basically I have stolen Andy's idea about using multi stage docker build.
https://docs.docker.com/develop/develop-images/multistage-build/

I think what we need here is exactly the use case for multistage builds, and as it is expected, the size of the base image is only increased by the size of the binary added to it (no overhead):
```

registry.access.redhat.com/ubi8-minimal   latest              c58d4cf08dc2        8 days ago          142MB
quay.io/nicob87/service-controller        multi               1ae6453d76df        36 seconds ago      183MB

$ docker run quay.io/nicob87/service-controller:multi ls -lh service-controller
-rwxr-xr-x. 1 root root 39M Jun 30 17:47 service-controller
```

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -16,28 +16,16 @@ build-site-controller:
...
 	${DOCKER} build -t ${SITE_CONTROLLER_IMAGE} -f Dockerfile.site-controller .
 
 docker-push:
 	${DOCKER} push ${SERVICE_CONTROLLER_IMAGE}
 	${DOCKER} push ${SITE_CONTROLLER_IMAGE}
 
-format:
-	go fmt ./...
-
-client-mock-test:
```
Comment (grs) [inline Makefile:30]: Did you intentionally remove these makefile targets? Seems unrelated to the purpose of the PR.

Comment (nicob87) [reply Makefile:30]: No!
Fixing it now

Comment (nicob87) [reply Makefile]: The faster reviewer in the far west!
